app/api/packets/helpers.ts

```diff
@@ -211,3 +211,31 @@ export function generateQueryParams(params: QueryFilters): string {
   if (params.where) reqParams += `where: {${params.where}}`;
   return '(' + reqParams + ')';
 }
+
+export interface FiltersProps {
+  start?: string,
+  end?: string,
+  states?: string,
+  src?: string,
+  dest?: string
+}
+
+export function generateSendPacketFilters(filters: FiltersProps): string {
+  let startFilter = filters.start ? `blockTimestamp_gte: ${filters.start}` : '';
+  const endFilter = filters.end ? `blockTimestamp_lte: ${filters.end}` : '';
+  if (startFilter && endFilter) { startFilter += ', '; }
+  let stateFilter = filters.states ? `packetRelation: {state_in: ${filters.states}}` : '';
+  if (stateFilter && (startFilter || endFilter)) { stateFilter = ', ' + stateFilter; }
+  let srcFilter = filters.src ? `portId_contains: ${filters.src}` : '';
+  const destFilter = filters.dest ? `counterpartyPortId_contains: ${filters.dest}` : '';
+  if (srcFilter && destFilter) { srcFilter += ', '; }
+  let channelFilter = '';
+  if (srcFilter || destFilter) {
+    channelFilter = `sourceChannel: {AND: {${srcFilter}${destFilter}}}`;
+  }
+  if (channelFilter && (startFilter || endFilter || stateFilter)) {
+    channelFilter = ', ' + channelFilter
+  }
+
+  return startFilter + endFilter + stateFilter + channelFilter;
+}
```

Comment:
This function is hard to reason about it. Can we simplify it?

Here are some ideas:
```
export function generateSendPacketFilters(filters: FiltersProps): string {
  const filterParts = [
    filters.start && `blockTimestamp_gte: ${filters.start}`,
    filters.end && `blockTimestamp_lte: ${filters.end}`,
    filters.states && `packetRelation: {state_in: ${filters.states}}`,
    (filters.src || filters.dest) && `sourceChannel: {AND: {
      ${filters.src ? `portId_contains: ${filters.src}` : ''}
      ${filters.src && filters.dest ? ',' : ''}
      ${filters.dest ? `counterpartyPortId_contains: ${filters.dest}` : ''}
    }}`
  ].filter(Boolean);

  return filterParts.join(', ');
}
```
or
```
export function generateSendPacketFilters(filters: FiltersProps): string {
  const startFilter = filters.start ? `blockTimestamp_gte: ${filters.start}` : '';
  const endFilter = filters.end ? `blockTimestamp_lte: ${filters.end}` : '';
  const stateFilter = filters.states ? `packetRelation: {state_in: ${filters.states}}` : '';
  const srcFilter = filters.src ? `portId_contains: ${filters.src}` : '';
  const destFilter = filters.dest ? `counterpartyPortId_contains: ${filters.dest}` : '';

  const channelFilter = (srcFilter || destFilter) 
    ? `sourceChannel: {AND: {${srcFilter}${srcFilter && destFilter ? ', ' : ''}${destFilter}}}`
    : '';

  return [
    startFilter, 
    endFilter, 
    stateFilter && `${startFilter || endFilter ? ', ' : ''}${stateFilter}`, 
    channelFilter && `${startFilter || endFilter || stateFilter ? ', ' : ''}${channelFilter}`
  ].filter(Boolean).join('');
}
```